templates/editAll.html

```diff
@@ -0,0 +1,252 @@
+$def with (selectedAY, fullMountingPlan)  
+
+$ title_string = 'Edit All Mountings and Quotas'
+
+$var title:$title_string
+$ home = ['/', 'Home']
+$ editAll = ['#', title_string]
+
+$var hierarchy = [home, editAll]
+
+<script type="text/javascript">
+    var selectedModuleCount = 0;
+
+    $$(document).ready(function(){
+        //Load the list of modules into the 'Select module to edit' input
+        var moduleSource = [];
+        var moduleCodeAndName = "";
+        $for subplan in fullMountingPlan:
+            moduleCodeAndName = "$subplan[0]"+" - "+"$subplan[1]"
+            moduleSource.push(moduleCodeAndName);
+        $$("input.typeahead").typeahead({
+            minLength: 1,
+            source: moduleSource,
+            updater: function(selectedModule) {
+                var selectedModuleCode = selectedModule.split(" ")[0];
+                var isModuleEdited = selectedModuleCode + "_isEdited";
+                showModuleInTable(selectedModuleCode, isModuleEdited);
+            }
+        })
+    });
+
```

Comment:
Your implementation seems to be showing/hiding module rows, then iterating through the full list of modules and making the edit if the edited flag is set. This can be expensive in the long run.

You could first store a list of modules and their associated data when this page loads, then only pass back those data which were modified back to the backend handler.